test/unit/eigensolver/mc/test_backtransformation.cpp

```diff
@@ -0,0 +1,169 @@
+using namespace dlaf::matrix::internal;
+using namespace dlaf::matrix::test;
+using namespace dlaf::test;
+using dlaf::matrix::test::MatrixLocal;
+using namespace testing;
+
+template <typename Type>
+class BackTransformationEigenSolverTestMC : public ::testing::Test {};
+TYPED_TEST_SUITE(BackTransformationEigenSolverTestMC, MatrixElementTypes);
+
+GlobalElementSize globalTestSize(const LocalElementSize& size) {
+  return {size.rows(), size.cols()};
+}
+
+const std::vector<std::tuple<SizeType, SizeType, SizeType, SizeType>> sizes =
+    // m, n, mb, nb
+    {{3, 0, 1, 1},   {0, 5, 2, 3},  // m, n = 0
+     {2, 2, 3, 3},   {3, 4, 6, 7},  // m < mb
+     {3, 3, 1, 1},   {4, 4, 2, 2}, {6, 3, 3, 3}, {12, 2, 4, 4}, {12, 24, 3, 3},
+     {24, 36, 6, 6}, {5, 8, 3, 2}, {4, 6, 2, 3}, {5, 5, 2, 3},  {8, 27, 3, 4},
+     {15, 34, 4, 6}, {7, 6, 2, 3}, {8, 5, 2, 3}, {7, 5, 2, 3}};
+
+template <class T>
+MatrixLocal<T> makeLocal(const Matrix<const T, Device::CPU>& matrix) {
+  return {matrix.size(), matrix.distribution().blockSize()};
+}
+
+template <class T>
+void set_zero(Matrix<T, Device::CPU>& mat) {
+  dlaf::matrix::util::set(mat, [](auto&&) { return static_cast<T>(0.0); });
+}
+
+template <class T>
+void getTau(T& tau, T dotprod, BaseType<T> /*tau_i*/) {
+  tau = static_cast<T>(2.0) / dotprod;
+}
+
+template <class T>
+void getTau(std::complex<T>& tau, T dotprod, BaseType<T> tau_i) {
+  tau = {(static_cast<T>(1.0) + sqrt(static_cast<T>(1.0) - dotprod * tau_i * tau_i)) / dotprod, tau_i};
+}
+
+template <class T>
+void testBacktransformationEigenv(SizeType m, SizeType n, SizeType mb, SizeType nb) {
+  LocalElementSize sizeC(m, n);
+  TileElementSize blockSizeC(mb, nb);
+  Matrix<T, Device::CPU> mat_c(sizeC, blockSizeC);
+  dlaf::matrix::util::set_random(mat_c);
+
+  LocalElementSize sizeV(m, m);
+  TileElementSize blockSizeV(mb, mb);
+  Matrix<T, Device::CPU> mat_v(sizeV, blockSizeV);
+  dlaf::matrix::util::set_random(mat_v);
+
+  if (m == 0 || n == 0)
+    return;
+
+  const SizeType nr_reflector = std::max(static_cast<SizeType>(0), m - mb - 1);
+
+  // Reset diagonal and upper values of V
+  MatrixLocal<T> v({m, m}, blockSizeV);
+  for (const auto& ij_tile : iterate_range2d(v.nrTiles())) {
+    // copy only the panel
+    const auto& source_tile = mat_v.read(ij_tile).get();
+    if (ij_tile.row() <= ij_tile.col()) {
+      tile::set0<T>(v.tile(ij_tile));
+    }
+    else if (ij_tile.row() == ij_tile.col() + 1) {
+      copy(source_tile, v.tile(ij_tile));
+      tile::laset<T>(lapack::MatrixType::Upper, 0.f, 1.f, v.tile(ij_tile));
+    }
+    else {
+      copy(source_tile, v.tile(ij_tile));
+    }
+  }
+
+  // Create C local
+  MatrixLocal<T> c({m, n}, blockSizeC);
+  for (const auto& ij_tile : iterate_range2d(c.nrTiles())) {
+    // copy only the panel
+    const auto& source_tile = mat_c.read(ij_tile).get();
+    copy(source_tile, c.tile(ij_tile));
+  }
+
+  common::internal::vector<hpx::shared_future<common::internal::vector<T>>> taus;

```

Comment:
`nr_reflector` is not the right value, since this is the vector of blocks, while this line is going to reserve space for the total number of taus ELEMENTS.

Reply:
sorry.. is that correct now? `nr_reflectors * mb`?

Reply:
https://www.tio.ch/svizzera/attualita/1525837/secondino-armadio-direttore-due-coperte

Reply:
No, it should be the number of blocks (as discussed for the assert).